src/components/snackbar/snackbar.tsx

```diff
@@ -99,89 +102,92 @@ export class Snackbar {
     @Event()
     private hide: EventEmitter<void>;
 
-    private mdcSnackbar: MDCSnackbar;
+    @State()
+    private offset: number = 0;
 
-    constructor() {
-        this.handleMdcClosing = this.handleMdcClosing.bind(this);
-    }
+    @State()
+    private open: boolean = false;
 
-    public connectedCallback() {
-        this.initialize();
-    }
+    @State()
+    private closing: boolean = true;
 
-    public componentDidLoad() {
-        this.initialize();
-    }
-
-    private initialize() {
-        const element = this.host.shadowRoot.querySelector('.mdc-snackbar');
-        if (!element) {
-            return;
-        }
-
-        this.mdcSnackbar = new MDCSnackbar(element);
-
-        this.mdcSnackbar.listen('MDCSnackbar:closing', this.handleMdcClosing);
-    }
-
-    public disconnectedCallback() {
-        this.mdcSnackbar.unlisten('MDCSnackbar:closing', this.handleMdcClosing);
-        this.mdcSnackbar.destroy();
+    @Listen('changeOffset')
+    protected onChangeIndex(event: CustomEvent<number>) {
+        event.stopPropagation();
+        this.offset = event.detail;
     }
 
     /**
      * Show the snackbar
      */
     @Method()
     public async show() {
-        if (this.timeout) {
-            this.mdcSnackbar.timeoutMs = this.timeout;
+        if (this.open) {
+            return;
         }
 
-        this.mdcSnackbar.labelText = this.message;
+        this.open = true;
+        this.closing = false;
+        container.add(this.host);
 
-        this.mdcSnackbar.open();
+        if (this.timeout) {
+            setTimeout(
+                this.close,
+                Math.max(this.timeout - hideAnimationDuration, 0),
+            );
+        }
     }
 
+    private close = () => {
```

Comment:
Since this is private, I'd prefer to move it below the render function.

Although, for the sake of symmetry, we might want to make this a public `@method` too?

Reply:
I would prefer not to have any `@Method` at all and instead have a property such as `open`, but that would be a breaking change so I left it out for now